.github/workflows/beam_Release_Python_NightlySnapshot.yml

```diff
@@ -0,0 +1,101 @@
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Release Nightly Snapshot Python
+
+on:
+  schedule:
+    - cron: '15 12 * * *'
+  workflow_dispatch:
+
+#Setting explicit permissions for the action to avoid the default permissions which are `write-all` in case of pull_request_target event
+permissions:
+  actions: write
+  pull-requests: read
+  checks: read
+  contents: read
+  deployments: read
+  id-token: none
+  issues: read
+  discussions: read
+  packages: read
+  pages: read
+  repository-projects: read
+  security-events: read
+  statuses: read
+
+env:
+  GRADLE_ENTERPRISE_ACCESS_KEY: ${{ secrets.GE_ACCESS_TOKEN }}
+  GRADLE_ENTERPRISE_CACHE_USERNAME: ${{ secrets.GE_CACHE_USERNAME }}
+  GRADLE_ENTERPRISE_CACHE_PASSWORD: ${{ secrets.GE_CACHE_PASSWORD }}
+
+jobs:
+  beam_Release_Python_NightlySnapshot:
+    name: ${{matrix.job_name}}  (${{matrix.job_phrase}})
+    runs-on: [self-hosted, ubuntu-20.04, main]
+    strategy:
+      matrix:
+        job_name: [beam_Release_Python_NightlySnapshot]
+        job_phrase: [Release Nightly Snapshot Python]
+    if: github.event_name == 'workflow_dispatch' ||
+        github.event_name == 'schedule'
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup repository
+        uses: ./.github/actions/setup-action
+        with:
+          github_job: ${{matrix.job_name}}
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          comment_phrase: ${{matrix.job_phrase}}
+      - name: Install Java
+        uses: actions/setup-java@v3.8.0
+        with:
+          distribution: 'zulu'
+          java-version: '8'
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2
+        with:
+          cache-read-only: false
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
+      - name: Authenticate on GCP
+        uses: google-github-actions/setup-gcloud@v0
+        with:
+          service_account_email: ${{ secrets.GCP_SA_EMAIL }}
+          service_account_key: ${{ secrets.GCP_SA_KEY }}
+          project_id: ${{ secrets.GCP_PROJECT_ID }}
+          export_default_credentials: true
+      # Used because script is written with jenkins in mind
+      # TODO: Remove it once the jenkins job is disabled 
+      #       and the script is updated    
```

Comment:
I think we should just disable the jenkins job when we merge this in and do it all in one shot. Otherwise we'll end up with multiple competing snapshots, which isn't great for anyone who relies on them (I think the risk of missing a nightly snapshot is relatively lower impact).

In general, I think that should be our approach for jobs that have non-trivial side effects

Reply:
I have added a commit that removes the .groovy file for the Python Nightly Release job. Let me know if something else i needed

Reply:
Should we address this TODO as well? I'm also fine following up on it separately as long as we track it

Reply:
Forgot to note that but yes, i will take it separately, soon. Wanted to do it separate in the case something is wrong with this PR we can more easily revert. If we verify this is ok next PR will be that script

Reply:
Sounds good to me